harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -570,79 +958,81 @@ def _compute_validation_metrics(self) -> workload.Metrics:
-            if self.is_chief:
-                metrics = self.trial.evaluate_full_dataset(data_loader=self.validation_loader)
+            if self._is_chief:
+                metrics = self._trial.evaluate_full_dataset(data_loader=self.validation_loader)
 
                 if not isinstance(metrics, dict):
                     raise TypeError(
                         f"eval() must return a dictionary, got {type(metrics).__name__}."
                     )
 
                 metrics = pytorch._convert_metrics_to_numpy(metrics)
-                num_inputs = self.context.get_per_slot_batch_size() * len(self.validation_loader)
+                num_inputs = self._context.get_per_slot_batch_size() * len(self.validation_loader)
 
         metrics.update(
-            pytorch._convert_metrics_to_numpy(self.context.reduce_metrics(for_training=False))
+            pytorch._convert_metrics_to_numpy(self._context.reduce_metrics(for_training=False))
         )
 
-        if self.context.distributed.size > 1 and any(
+        if self._context.distributed.size > 1 and any(
             util.is_overridden(c.on_validation_end, pytorch.PyTorchCallback)
             or util.is_overridden(c.on_validation_step_end, pytorch.PyTorchCallback)
-            for c in self.callbacks.values()
+            for c in self._callbacks.values()
         ):
             logging.debug(
                 "Broadcasting metrics to all worker processes to execute a "
                 "validation step end callback"
             )
-            metrics = self.context.distributed.broadcast(metrics)
+            metrics = self._context.distributed.broadcast(metrics)
 
-        for callback in self.callbacks.values():
+        for callback in self._callbacks.values():
             if util.is_overridden(callback.on_validation_step_end, pytorch.PyTorchCallback):
                 logging.warning(
                     "on_validation_step_end is now deprecated, please use on_validation_end instead"
                 )
                 callback.on_validation_step_end(metrics)
 
-        for callback in self.callbacks.values():
+        for callback in self._callbacks.values():
             callback.on_validation_end(metrics)
 
-        if not self.is_chief:
-            return {}
+        self._state.last_val = self._state.batches_trained
 
-        # Skip reporting timings if evaluate_full_dataset() was defined.  This is far less common
-        # than evaluate_batch() and we can't know how the user processed their validation data.
-        if self._evaluate_batch_defined():
-            step_duration = time.time() - step_start_time
-            logging.info(
-                det.util.make_timing_log("validated", step_duration, num_inputs, num_batches)
-            )
-        self.metric_writer.on_validation_step_end(self.steps_completed, metrics)
-        return {"num_inputs": num_inputs, "validation_metrics": metrics}
```

Comment:
Why was this changed?  It looks like the old strategy could still apply, with less code churn (maybe no code churn?)

Reply:
done